modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractJavaCodegen.java

```diff
@@ -1421,6 +1437,39 @@ public void preprocessOpenAPI(OpenAPI openAPI) {
         }
     }
 
+    /**
+     * Add all OneOf schemas to #/components/schemas and replace them in the original content by ref schema
+     * Replace OneOf with unmodifiable types with an empty Schema
+     *
+     * OpenAPI Parser does not add inline OneOf schemas to models to generate
+     *
+     * @param content a 'content' section in the OAS specification.
+     */
+    private void repairInlineOneOf(final Content content) {
+        content.values().forEach(mediaType -> {
+            final Schema replacingSchema = mediaType.getSchema();
+            if (isOneOfSchema(replacingSchema)) {
+                if (ModelUtils.isSchemaOneOfConsistsOfCustomTypes(openAPI, replacingSchema)) {
+                    final String oneOfModelName = (String) replacingSchema.getExtensions().get("x-one-of-name");
+                    final Schema newRefSchema = new Schema<>().$ref("#/components/schemas/" + oneOfModelName);
+                    mediaType.setSchema(newRefSchema);
+                    ModelUtils.getSchemas(openAPI).put(oneOfModelName, replacingSchema);
+                } else {
+                    mediaType.setSchema(new Schema());
+                }
+            }
+        });
+    }
+
+    private static boolean isOneOfSchema(final Schema schema) {
+        if (schema instanceof ComposedSchema) {
+            ComposedSchema composedSchema = (ComposedSchema) schema;
+            return Optional.ofNullable(composedSchema.getProperties()).map(Map::isEmpty).orElse(true)
+                    && Optional.ofNullable(schema.getExtensions()).map(m -> m.containsKey("x-one-of-name")).orElse(false);
+        }
+        return false;
+    }
+
```

Comment:
FYI. Since these changes occur in the abstract java codegen, which is used by the java client codegen, we will need to test the Java client (jersey2, native) generator carefully which already supports oneOf implementation (not using interface though)

Have you tried the oneOf implementation in Java client (jersey2, native) generator? Does it meet your requirement?